flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2143,11 +2144,29 @@ def test_deleted_ignored(self):
                 metadata={u"foo": u"bar"},
             )
         )
+        foreign_registered_dataset_id = uuid4()
+        foreign_registered_dataset = DiscoveredDataset(
+            dataset_id=foreign_registered_dataset_id,
+            blockdevice_id=_create_blockdevice_id_for_test(
+                foreign_registered_dataset_id
+            ),
+            state=DatasetStates.REGISTERED,
+        )
 
         assert_calculated_changes(
             self, local_state, local_config,
             nonmanifest_datasets={},
             expected_changes=in_parallel(changes=[]),
+            discovered_datasets=[
+                foreign_registered_dataset
+            ],
+            persistent_state=PersistentState().transform(
```

Comment:
:question: +1 on getting this in a unit test. This file is full with tests that are sort of... tricky to maintain. There is a very big overhead to re-architecting this part of flocker. I don't think this makes the problem any worse, but it's something I'm always on the lookout for.